spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/Utils.scala

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.intel.analytics.bigdl.models.maskrcnn
+
+import breeze.linalg.{*, dim, max}
+import com.intel.analytics.bigdl.nn.ResizeBilinear
+import com.intel.analytics.bigdl.nn.abstractnn.DataFormat
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.sun.xml.internal.bind.v2.TODO
+
+import scala.collection.mutable.ArrayBuffer
+
+object Utils {
+  // box with 4 element (xyxy)
+  def expandBoxes(bbox: Tensor[Float], bboxExpand: Tensor[Float], scale: Float)
+  : Unit = {
+    require(bbox.nElement() == 4 && bboxExpand.nElement() == 4
+      && bbox.dim() == 1 && bboxExpand.dim() == 1,
+      "Box and expected Box should have 4 elements with one dim")
+
+    val box0 = bbox.valueAt(1)
+    val box1 = bbox.valueAt(2)
+    val box2 = bbox.valueAt(3)
+    val box3 = bbox.valueAt(4)
+
+    var wHalf = (box2 - box0) * 0.5f
+    var hHalf = (box3  - box1) * 0.5f
+    val x_c = (box2 + box0) * 0.5f
+    val y_c = (box3 + box1) * 0.5f
+
+    wHalf *= scale
+    hHalf *= scale
+
+    bboxExpand.setValue(1, x_c - wHalf)
+    bboxExpand.setValue(3, x_c + wHalf)
+    bboxExpand.setValue(2, y_c - hHalf)
+    bboxExpand.setValue(4, y_c + hHalf)
+  }
+
+  // mask with three dims (channel, height, wide)
+  def expandMasks(mask: Tensor[Float], padding: Int): (Tensor[Float], Float) = {
+    require(mask.isContiguous(), "Only support contiguous mask")
+
+    val N = mask.size(1)
```

Comment:
what does `N` `M` mean ? also `pad2` is not good naming

Reply:
ok, changed.